packages/deployments/contracts/contracts/lib/Connext/ConnextUtils.sol

```diff
@@ -3,29 +3,326 @@ pragma solidity 0.8.11;
+   * @param amount - The amount of liquidity added
+   * @param caller - The account that called the function
+   */
+  event LiquidityAdded(address indexed router, address local, bytes32 canonicalId, uint256 amount, address caller);
+
+  /**
+   * @notice Emitted when `bumpTransfer` is called by an user on the origin domain
+   * @param transferId - The unique identifier of the crosschain transaction
+   * @param relayerFee - The updated amount of relayer fee in native asset
+   * @param caller - The account that called the function
+   */
+  event TransferRelayerFeesUpdated(bytes32 indexed transferId, uint256 relayerFee, address caller);
+
+  /**
+   * @notice Emitted when `initiateClaim` is called on the destination chain
+   * @param domain - Domain to claim funds on
+   * @param recipient - Address on origin chain to send claimed funds to
+   * @param caller - The account that called the function
+   * @param transferIds - TransferIds to claim
+   */
+  event InitiatedClaim(uint32 indexed domain, address indexed recipient, address caller, bytes32[] transferIds);
+
+  /**
+   * @notice Emitted when `claim` is called on the origin domain
+   * @param recipient - Address on origin chain to send claimed funds to
+   * @param total - Total amount claimed
+   * @param transferIds - TransferIds to claim
+   */
+  event Claimed(address indexed recipient, uint256 total, bytes32[] transferIds);
+
+  // ============ Admin Functions ============
+
+  /**
+   * @notice Used to add an AMM for adopted <> local assets
+   * @param _canonical - The canonical TokenId to add (domain and id)
+   * @param _stableSwap - The address of the amm to add
+   */
+  function addStableSwapPool(
+    ConnextMessage.TokenId calldata _canonical,
+    address _stableSwap,
+    mapping(bytes32 => IStableSwap) storage _adoptedToLocalPools
+  ) external {
+    // Update the pool mapping
+    _adoptedToLocalPools[_canonical.id] = IStableSwap(_stableSwap);
+
+    emit StableSwapAdded(_canonical.id, _canonical.domain, _stableSwap, msg.sender);
+  }
+
+  /**
+   * @notice Used to add assets on same chain as contract that can be transferred.
+   * @param _canonical - The canonical TokenId to add (domain and id)
+   * @param _adoptedAssetId - The used asset id for this domain (i.e. PoS USDC for
+   * polygon)
+   */
+  function addAssetId(
+    ConnextMessage.TokenId calldata _canonical,
+    address _adoptedAssetId,
+    address _wrapper,
+    mapping(bytes32 => bool) storage _approvedAssets,
+    mapping(address => ConnextMessage.TokenId) storage _adoptedToCanonical,
+    mapping(bytes32 => address) storage _canonicalToAdopted
+  ) external {
+    // Sanity check: needs approval
+    if (_approvedAssets[_canonical.id]) revert ConnextUtils__addAssetId_alreadyAdded();
+
+    // Update approved assets mapping
+    _approvedAssets[_canonical.id] = true;
+
+    // Update the adopted mapping
+    _adoptedToCanonical[_adoptedAssetId] = _canonical;
```

Comment:
This has to switch to the supported asset before being keyed on anything. See require [here](https://github.com/connext/nxtp/blob/7c8d2c6bd2633cf47de32845d381bb381eed1428/packages/deployments/contracts/contracts/lib/Connext/ConnextUtils.sol#L774-L779)